util/src/keys/directory.rs

```diff
@@ -484,6 +492,10 @@ impl KeyDirectory {
 			let json_bytes = json_text.into_bytes();
 			try!(file.write(&json_bytes));
 		}
+		if let Err(error_code) = restrict_permissions_owner(self.key_path(&key_file.id).as_path()) {
+			fs::remove_file(self.key_path(&key_file.id)).unwrap();
+			panic!("fatal: failed to modify permissions of the file (chmod: {})", error_code);
```

Comment:
should be a warning, not a panic. Running with a read-only key store is a viable scenario